apps/api-extractor/src/collector/Collector.ts

```diff
@@ -296,41 +296,18 @@ export class Collector {
   }
 
   /**
-   * Returns the associated `CollectorEntity` for the given `astEntity`, if one was created during analysis.
-   */
-  public tryGetCollectorEntity(astEntity: AstEntity): CollectorEntity | undefined {
-    return this._entitiesByAstEntity.get(astEntity);
-  }
-
-  /**
-   * Returns the name to emit for the given `symbol`.
-   * This is usually the `name` of that `symbol`.
-   * Only if this `symbol` has a `CollectorEntity`, the `nameForEmit` of that entity is used.
-   * @param symbol the symbol to compute the name to emit for
-   * @returns the name to emit for the given `symbol`
+   * For a given analyzed ts.Symbol, return the CollectorEntity that it refers to. Returns undefined if it
+   * doesn't refer to anything interesting.
    */
-  public getEmitName(symbol: ts.Symbol): string {
-    const collectorEntity: CollectorEntity | undefined = this._entitiesBySymbol.get(symbol);
-    return collectorEntity?.nameForEmit ?? symbol.name;
+  public tryGetEntityForSymbol(symbol: ts.Symbol): CollectorEntity | undefined {
+    return this._entitiesBySymbol.get(symbol);
   }
 
   /**
-   * Returns (the symbol of) the namespace that exports the given `symbol`, if applicable.
-   * @param symbol the symbol representing the namespace that exports the given `symbol`,
-   *   or undefined if there is no such namespace
+   * Returns the associated `CollectorEntity` for the given `astEntity`, if one was created during analysis.
    */
-  public getExportingNamespace(symbol: ts.Symbol): ts.Symbol | undefined {
-    const collectorEntity: CollectorEntity | undefined = this._entitiesBySymbol.get(symbol);
-    if (collectorEntity) {
-      const exportingNamespace: AstNamespaceImport | undefined = collectorEntity.getExportingNamespace();
-      if (exportingNamespace) {
-        return TypeScriptInternals.tryGetSymbolForDeclaration(
-          exportingNamespace.declaration,
-          this.typeChecker
-        );
-      }
-    }
-    return undefined;
```

Comment:
I moved the bulk of this logic as well as the `CollectorEntity.getExportingNamespace` method to `DeclarationReferenceGenerator`. It seemed fairly bespoke and I didn't see an obvious way to generalize it such that it made sense to live in the `Collector` (i.e. working with `ts.Symbols` and such).

Reply:
Yes, this distribution of responsibilities seems better to me, too.